### DISCUSSION TITLE
July Newsletter

### DISCUSSION THREAD

**Discussion by briancl2** [briancl2/CustomerNewsletter#4] [Customer Newsletter]
This is a personally curated newsletter for just my customers. The purpose is to share new resources, announcements, and events. I welcome your feedback, and please let me know if you would like to add or remove anyone from this list.

Highlights for July are Copilot Enterprise Mixed Licensing & other enhancements, Updated ADO + GHE Guidance, and a GenAI Industry Roundup

### Copilot Enterprise Mixed Licensing & More
At GA launch earlier in the year, Copilot Enterprise customers faced an all-or-nothing choice: either enroll in Copilot Business or Copilot Enterprise. As of [July 26](https://github.blog/changelog/2024-07-26-copilot-enterprise-mixed-licensing-beta/), it is now possible for a single GHE account to mix both Copilot Business and Copilot Enterprise licenses. This could be particularly helpful for organizations that only partially migrated into GitHub repos which remains a pre-requisite for using Copilot Enterprise. 

Also in July, Copilot Enterprise saw many more updates: 
* [github.com skills](https://github.blog/changelog/2024-07-05-github-copilot-enterprise-on-gpt-4o) and [Copilot Chat](https://github.blog/changelog/2024-07-31-github-copilot-chat-and-pull-request-summaries-are-now-powered-by-gpt-4o/) were upgraded to GPT-4o for faster performance and more recent training data. Copilot Business also received the same model upgrade for Chat in the IDE.
* [text completion while you type pull request descriptions](https://github.blog/changelog/2024-07-24-copilot-text-completion-for-pull-request-descriptions-beta) to speed up administrative tasks
* Copilot Chat now [knows about pull requests, discussions, and files](https://github.blog/changelog/2024-06-27-copilot-enterprise-knows-about-pull-requests-discussions-and-files-june-updates) to give Copilot Enterprise users additional use cases and greater context
* [Visual Studio Copilot Chat is now enriched with knowledge bases](https://github.blog/changelog/2024-07-09-copilot-knowledge-bases-in-visual-studio-preview) from github.com to provide more customized and accurate responses to developer’s IDE Chat sessions.

### The future of Azure DevOps & the GitHub Platform. What is "Better Together?"
Customers paying attention to the ADO and GHE roadmaps are curious to know how to plan for the future. The simplest way to summarize those roadmaps: one product family has significant investment in AI-assisted development and security, Actions, Issues, Projects, Data Residency, etc, and the other product family is prioritizing enhancing integrations with GitHub. The direction should be clear, however we aren’t able to share any dates for major product changes, yet.

Those enhanced GitHub integrations, including a [July 25 update for ADO Boards + GitHub integration](https://devblogs.microsoft.com/devops/update-on-azure-boards-github-integration/), will help smooth the transition from Full ADO to Full GHE by allowing users to move in phases. E.g., product teams doing active development on ADO can start by moving their codebase into a GHE repo and continue to use ADO Boards and Pipelines. With just a repo migration, customers unlock all of the Copilot Enterprise capabilities and GHAS with Copilot Autofix without having to leave Boards or Pipelines.

The recommendation for those customers is to begin the repo-only migration for existing product teams on ADO and establish a directive to start all net new development projects on the full GHE suite. This is a pragmatic way to minimize future migration work and avoid the disruption of a one-shot migration. If you’d like to see the ADO + GitHub integrations in action or find out more about the recommended migration path, reply to this email and we will set that up.

### GenAI in Software Development Industry Roundup 
* Engineering Enablement Podcast: [How to measure GenAI adoption and impact in software development](https://getdx.com/podcast/how-to-measure-genai-adoption-impact/). This is an information dense 30 minute podcast that covers a lot of real world experience from engineering leaders at Airbnb, Jumio, and GitHub. Topics include:
  - Challenges of measuring AI productivity
  - Use cases for GenAI in software development
  - GitHub's process for testing new GenAI tools
  - GenAI Adoption Strategies at Airbnb
  - Productivity Gains at Airbnb
  - Developer survey process and results at Jumio
  - Biggest Gains of GenAI at Airbnb
* The Pragmatic Engineer published [AI Tooling for Software Engineers in 2024: Reality Check](https://newsletter.pragmaticengineer.com/p/ai-tooling-2024). It contains updated data and direct quotes from over 200 diverse developers across a number of industries, company sizes, and experience levels. The article is free, but one section is pay-walled to subscribers.
* Rob Bos, a GitHub Trainer at the partner Xebia, posted a clear and concise guide for getting the most out of GenAI in software development, [GitHub Copilot - Levels of Enlightenment](https://devopsjournal.io/blog/2024/06/07/GitHub-Copilot-Levels-of-enlightenment). Rob also included a companion [YouTube Playlist](https://www.youtube.com/playlist?list=PLXVVwOM8uv2y0Yo6H8qu9giWWWlZLzu8K).

### Other GitHub Copilot resources
* See the first detailed look at GitHub’s upcoming Copilot Extensions from [Atlasian, Elastic, and Split’s joint webinar](https://www.youtube.com/watch?v=JclDSHrSZV0) recorded in July.
* Copilot Individual usage (i.e., personal Copilot accounts that fall outside the enterprise policies) can now be [blocked at the network level](https://github.blog/changelog/2024-07-31-copilot-network-requests-are-now-routed-based-on-subscription/) to ensure all Copilot users are adhering to enterprise policies
* Brian’s personally curated Copilot Training & Tips [Youtube playlist](https://www.youtube.com/playlist?list=PLCiDM8_DsPQ1WJ5Ss3e0Lsw8EaijUL_6D), updated monthly

### Upcoming Webinars
* August 7: [GitHub Roadmap Q3](https://resources.github.com/webcasts/github-roadmap-webinar-q3/)
* August 21: [How AI Can Scale AppSec - GitHub and Endor Labs](https://developer.microsoft.com/en-us/reactor/series/S-1311/)
* August 28: [GitHub Copilot Live AMA](https://www.youtube.com/watch?v=5drqCXv16js)

### Upcoming In-person events
* Las Vegas - August 7-8: [GitHub is at Black Hat USA](https://resources.github.com/github-black-hat-usa-2024/)
* San Francisco - August 20: [Microsoft AI Innovation Summit](https://developer.microsoft.com/en-us/reactor/events/22772/)
* NYC - September 12: NY Yankees Suite with Endor Labs
* Chicago - September 19: Chicago Cubs Rooftop
* San Francisco - Oct 29-30: [Discounted Tickets](https://githubuniverse.com/#tickets) available through September 3

**Comment by alwell-kevin** [comment]
Thank you for this thoughtful roundup! Always a pleasure to consume.
